### ISSUE TITLE
how to use custom SwAV backbone (finetuned on my own dataset with SSL) 

### ISSUE THREAD

**Issue by vkocaman** [facebookresearch/Detic#16]
Hi,

First of all, this is great !! thanks for the great work..

Is it possible to use custom SwAV backbone that in already finetuned on my own dataset with SSL. Can we just replace the ResNet block within Detectron with the one I finetuned with SwAV ? I assume we'd need to retrain the whole Detectron from scratch ?

What I basically want to do is to, making a particular class (that are already underrepresented in all the datasets used in Detic) much stronger by adding some more labels (segmentations) or just train SwAV with my unlabelled dataset from that particular classes, and then use this inside Detic to see it would do better than what it already does.

**Comment by xingyizhou** [comment]
Hi,
  Sorry for the delayed response. I am not sure if I get the full question.

  - If you only want to detect/ segment a particular class, I feel it makes more sense to just train on that class (or plus related/ similar classes), instead of training on all classes in Detic. You can fine-tune on ImageNet images for that class only on your pretraining use Detic losses. Or you can take our trained Detic models and fine-tune on your own dataset (instead of pertaining from SwAV)
  - If you want to improve the performance of all classes in Detic, using another pertaining might not be ideal, as we are already using a strong pertaining model.

**Comment by vkocaman** [comment]
@xingyizhou thx for tour answer. Could you please point me to a script or approach that I can use to fine tune on my segmentation dataset using one of your Detic model weights ? Is this like finetuning Detr ? Any recent notebook or blog that explains that ? Could you please advise ?

Thanks

**Comment by xingyizhou** [comment]
You can check the detectron2 tutorial [here](https://detectron2.readthedocs.io/en/latest/tutorials/datasets.html) for fine-tuning on custom datasets.

**Comment by vkocaman** [comment]
@xingyizhou thx for your respond.. actually I already did that.. but I used Detic model wieghts with the corresponding yaml file to fine tune on my dataset in coc format, using `detectron2`. The result was not good though. Even worse than what I get with default detectron2 checkpoints (`faster_rcnn_R_50_FPN_3x.yaml`). Here is my settings:

```
from detectron2.engine import DefaultTrainer
from detectron2.config import get_cfg

cfg = get_cfg()
cfg.merge_from_file='/content/Detic_LCOCOI21k_CLIP_SwinB_896b32_4x_ft4x_max-size.yaml'
#https://raw.githubusercontent.com/facebookresearch/Detic/main/configs/Detic_LCOCOI21k_CLIP_SwinB_896b32_4x_ft4x_max-size.yaml
cfg.DATASETS.TRAIN = ("my_dataset_train_6",)
cfg.DATASETS.TEST = ()
cfg.DATALOADER.NUM_WORKERS = 2
cfg.MODEL.WEIGHTS = 'https://dl.fbaipublicfiles.com/detic/Detic_LCOCOI21k_CLIP_SwinB_896b32_4x_ft4x_max-size.pth'
cfg.SOLVER.IMS_PER_BATCH = 8
cfg.SOLVER.BASE_LR = 0.00025  
cfg.SOLVER.MAX_ITER = 600   
cfg.SOLVER.STEPS = []       
cfg.MODEL.ROI_HEADS.BATCH_SIZE_PER_IMAGE = 128  
cfg.MODEL.ROI_HEADS.NUM_CLASSES = 1  
os.makedirs(cfg.OUTPUT_DIR, exist_ok=True)
trainer = DefaultTrainer(cfg) 
trainer.resume_or_load(resume=False)
trainer.train()
```

**Comment by xingyizhou** [comment]
Hi,
  Here are a few issues I noted:
- A `DefaultTrainer` will load the dataset correctly. You will need to use our customized dataloader. See our [train_net.py](https://github.com/facebookresearch/Detic/blob/main/train_net.py#L130).
- `cfg.MODEL.ROI_HEADS.NUM_CLASSES = 1` might not work (I am expecting an error here). You will need to generate the CLIP embeddings following [here](https://github.com/facebookresearch/Detic/tree/main/datasets#metadata).
- `BASE_LR` should be `0.000025` if you follow the linear learning rate rule.
